controllers/student.go

```diff
@@ -153,3 +175,48 @@ func StudentBlogLink(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	fmt.Fprint(w, "BlogLink successfully updated.")
 }
+func CreateStudentDashboard(modelStudent models.Student, db *gorm.DB) StudentDashboard {
+	var projects []ProjectDashboard
+	for _, proj_id := range strings.Split(modelStudent.ProjectsWorked, ",") {
+		var project ProjectDashboard
+		db.Table("projects").
+			Where("id = ?", proj_id).
+			Select("name", "repo_link").
+			First(&project)
+		projects = append(projects, project)
+	}
+	languages := strings.Split(modelStudent.LanguagesUsed, ",")
+	return StudentDashboard{
+		Name:           modelStudent.Name,
+		Username:       modelStudent.Username,
+		College:        modelStudent.College,
+		PassedMidEvals: modelStudent.PassedMidEvals,
+		PassedEndEvals: modelStudent.PassedEndEvals,
+		CommitCount:    modelStudent.CommitCount,
+		PullCount:      modelStudent.PullCount,
+		LinesAdded:     modelStudent.LinesAdded,
+		LinesRemoved:   modelStudent.LinesRemoved,
+		LanguagesUsed:  languages,
+		ProjectsWorked: projects,
+	}
+}
+
+func FetchStudentDashboard(w http.ResponseWriter, r *http.Request) {
+	app := r.Context().Value(middleware.APP_CTX_KEY).(*middleware.App)
+	db := app.Db
+
+	var modelStudent models.Student
+
+	login_username := r.Context().Value(middleware.LoginCtxKey(middleware.LOGIN_CTX_USERNAME_KEY))
+	tx := db.
+		Table("students").
+		Where("username = ?", login_username).
+		First(&modelStudent)
+
+	if tx.Error != nil {
```

Comment:
Can you handle the record not found error separately and return a `student does not exist` error in that case?